desc/compute/utils.py

```diff
@@ -77,46 +78,46 @@ def compute(  # noqa: C901
     bad_kwargs = kwargs.keys() - allowed_kwargs
     if len(bad_kwargs) > 0:
         raise ValueError(f"Unrecognized argument(s): {bad_kwargs}")
+    if not jitable:
```

Comment:
`CoilSet` objects have logic to deal with computing its tree structure, which would be annoying to implement at the objective level every time in order to get to the base Coil object that can then be called with _compute. Having a jitable kwarg here to skip the unneeded checks lets us use coilset.compute inside of objectives. 

This is due to an uncaught bug in `CoilObjectives`, any objective that requires compute quantitiy that needs a transform with derivatives to be built (like `"x"` `"x_s"`), the `has_transforms` check will fail due to a JAX error on us calling `to_list` on the derivatives. I can post the exact error later

Reply:
OK, for an immediate fix it sounds like it might be better to just mark `Transform._derivs` as a static attribute?

Bigger picture though it would be nice to have the option to ignore checks when calling `compute`, this is why we call `_compute` in all the other objectives but that's kind of hacky. We could move all these checks to their own function, and then compute would be something like

```python
def compute(..., check=True):
    if check:
        _check_inputs(...)
    return _compute(...)
```

Then we can use `compute` everywhere safely, just sometimes needing to pass in `check=False`

Reply:
> This is due to an uncaught bug in CoilObjectives, any objective that requires compute quantitiy that needs a transform with derivatives to be built (like "x" "x_s"), the has_transforms check will fail due to a JAX error on us calling to_list on the derivatives. I can post the exact error later

I'm confused by this, don't objectives like `CoilCurvature` and `CoilTorsion` require derivatives of the position? Why do they not fail?

Reply:
I also was confused, I think it is somehow bc the target compute quantitiy itself has no transforms/deriv requirements, only the dependencies. I did not dig much deeper into it though